src/pages/Register.vue

```diff
@@ -36,17 +39,25 @@ const newUser = ref<User>({
 });
 
 function saveUser() {
-  addUser(newUser.value);
+  errors.value = addUser(newUser.value)
+  if (errors.value.length === 0) {
+    loginUser(newUser.value.email, newUser.value.password)
+    router.push("/");
+  }
 }
 </script>
 <template>
   <div class="d-flex justify-content-center align-items-center" style="min-height: 100vh">
     <div class="card w-auto">
       <div class="card-body">
+        <div class="alert alert-danger" role="alert" v-if="errors.length">
+          <div v-for="error in errors">{{ error }}</div>
+        </div>
+
         <form @submit.prevent="saveUser">
-          <input type="text" class="form-control mb-3" placeholder="Benutzername" v-model="newUser.username" />
-          <input class="form-control mb-3" type="password" placeholder="Passwort" v-model="newUser.password" />
-          <input class="form-control mb-2" type="text" placeholder="E-Mail-Adresse" v-model="newUser.email" />
+          <input type="text" class="form-control mb-3" placeholder="Benutzername" v-model="newUser.username" required />
+          <input class="form-control mb-3" type="password" placeholder="Passwort" v-model="newUser.password" required />
+          <input class="form-control mb-2" type="text" placeholder="E-Mail-Adresse" v-model="newUser.email" required />
           <div class="">
```

Comment:
```suggestion
          <div>
```
Ich glaube, das war noch von mir übrig.